tests/integration/test_charm.py

```diff
@@ -14,13 +15,12 @@ async def test_build_and_deploy(ops_test):
     resources = {"oci-image": "quay.io/dexidp/dex:v2.28.1"}
     await ops_test.model.deploy(my_charm, resources=resources)
     await ops_test.model.wait_for_idle()
+    # Dex auth passes to active from idle again before passing completely to active.
+    sleep(5)
```

Comment:
I'm not a big fan of arbitrary sleeps in tests, since a randomly slow host could cause failures. I'm also a little confused about the comment here. "active" is a workload status and "idle" is an agent status; did you perhaps mean "executing" instead of "active"?  If so, what causes it to be idle for > 15s before going back to executing, and is the workload status "active" during that period or something else?  If it's "active", then you could use `wait_for_idle(wait_for_active=True)` to make it consider "idle but not active" as still pending.